MobilityData/mobility-feed-api
# feat: add backup and retention period and cloud storage soft delete policy

**Summary:**

Changes:
 - Setup backup to be taken daily with a retention period of 31 days for prod and 1 day for pre-prod
 -  Enable point-in-time recovery (prod) with 7 days for prod and 1 day for pre-prod environments.
 - Set instance delete protection to true (dev, qa, and prod)
 - Set a soft delete policy for the datasets' GCP bucket to 31 days in prod and 1 day for pre-prod environments
 - Upgraded google/terraform provider to `5.7.0` to support soft delete functionalities.
 - Updated the [Data Management Plan](https://docs.google.com/document/d/1keMMNXCklNNENk-hRhHQc4uGSopRhX5C1tJHpaVCTSw/edit?usp=sharing) document.

**Expected behavior:** 

All changes are applied to QA

**Testing tips:**

Soft delete policy
- Go to the GCP console dashboard in QA
- Visit the mobilitydata-datasets-qa bucket
- Create a folder and upload a file
- Delete the file
- Check the [Data Management Plan](https://docs.google.com/document/d/1keMMNXCklNNENk-hRhHQc4uGSopRhX5C1tJHpaVCTSw/edit?usp=sharing) and perform a recovery process as documented

Review SQL instance settings
- Go to the GCP console dashboard in QA
- Visit the SQL service in QA
- Verify all settings matches with the expected values

Closes #468 , #469
Please make sure these boxes are checked before submitting your pull request - thanks!
